### DISCUSSION TITLE
Styles still rendering when using NewCSSMiddleware?

### DISCUSSION THREAD

**Discussion by Linkinlog** [a-h/templ#659] [Q&A] [answered]
I think I might be a little confused on how the css is meant to be handled in Templ (v0.2.648). 

I am using css class expressions as per the docs like so
```css
css HeaderBrand() {
	font-weight: bold;
}
```
and applying them like so `class={ HeaderBrand() }`

For routing I am just using `net/http`, I have a map and corresponding slice containing all of my class expressions ( `s.classes` in the example ) and am wrapping the router with the middleware function like so
```go
mux := http.NewServeMux()

mux.HandleFunc("GET /", handleLanding)
...

handler := templ.NewCSSMiddleware(mux, s.classes...)

server := &http.Server{
    Addr:    s.addr,
    Handler: handler,
}

return server.ListenAndServe()
```

Yet for some reason I am still seeing `<style>` tags accompanying all my components, even though I can see the same styles in the new `/styles/templ.css` file. 

Any help is much appreciated!

**Comment by joerdav** [comment]
Please can you share the code of your http handler where you render the template? Just for more context?

**Reply by Linkinlog** [reply]
Of course

```go
func handleLanding(w http.ResponseWriter, r *http.Request) {
	if r.URL.Path != "/" {
		handleNotFound(w, r)
	}
	p := web.NewPage("Landing", "Welcome to the landing page")

	p.Layout(web.Landing()).Render(context.Background(), w)
}

```

and heres the details on `p`

```go
func NewPage(title, description string) *Page {
	url := "https://example.org"
	img := "https://example.org/assets/logo.webp"
	return &Page{
		Title:       title,
		Description: description,
		Img:         img,
		URL:         url,
	}
}

type Page struct {
	Title       string
	Description string
	Img         string
	URL         string
}
```

**Reply by a-h** [reply, marked as answer]
I can see the issue now.

The way that the CSS middleware determines whether the CSS has already been rendered or not, is by setting the HTTP context.

However, you're passing `context.Background()` instead of passing the `r.Context()` down through the template `Render` function.

**Reply by Linkinlog** [reply]
That did it! Thanks for the help
